tests/kernel/interrupt/src/nested_irq.c

```diff
@@ -1,170 +1,133 @@
 /*
+ * Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
  * Copyright (c) 2018 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-#include "interrupt.h"
 
-#define DURATION 5
-struct k_timer timer;
-struct k_timer irqlock_timer;
+#include <ztest.h>
+#include "interrupt_util.h"
 
-/* This tests uses two IRQ lines, selected within the range of IRQ lines
- * available on the target SOC the test executes on (and starting from
- * the maximum available IRQ line index)
- */
-#define IRQ_LINE(offset) (CONFIG_NUM_IRQS - ((offset) + 1))
+#define DURATION	5
 
-#define ISR0_OFFSET 1
-#define ISR1_OFFSET 2
+#define ISR0_TOKEN	0xDEADBEEF
+#define ISR1_TOKEN	0xCAFEBABE
 
-/* Keeping isr0 to be lowest priority than system timer
- * so that it can be interrupted by timer triggered.
- * In NRF5, RTC system timer is of priority 1 and
- * in all other architectures, system timer is considered
- * to be in priority 0.
+/*
+ * This test uses two IRQ lines selected within the range of available IRQs on
+ * the target SoC.  These IRQs are platform and interrupt controller-specific,
+ * and must be specified for every supported platform.
+ *
+ * In terms of priority, the IRQ1 is triggered from the ISR of the IRQ0;
+ * therefore, the priority of IRQ1 must be greater than that of the IRQ0.
  */
 #if defined(CONFIG_CPU_CORTEX_M)
-u32_t irq_line_0;
-u32_t irq_line_1;
-#define ISR0_PRIO 2
-#define ISR1_PRIO 1
+/*
+ * For Cortex-M NVIC, unused and available IRQs are automatically detected when
+ * when the test is run.
```

Comment:
double "when"

Reply:
I will fix that in another PR since this already got merged.

Reply:
Addressed in #23881